include/aws/crt/external/StringView.h

```diff
@@ -0,0 +1,819 @@
+}
+
+template <class _CharT, class _Traits>
+constexpr bool operator<=(
+    basic_string_view<_CharT, _Traits> __lhs,
+    typename std::common_type<basic_string_view<_CharT, _Traits>>::type __rhs) noexcept
+{
+    return __lhs.compare(__rhs) <= 0;
+}
+
+template <class _CharT, class _Traits>
+constexpr bool operator<=(
+    typename std::common_type<basic_string_view<_CharT, _Traits>>::type __lhs,
+    basic_string_view<_CharT, _Traits> __rhs) noexcept
+{
+    return __lhs.compare(__rhs) <= 0;
+}
+
+// operator >=
+template <class _CharT, class _Traits>
+constexpr bool operator>=(basic_string_view<_CharT, _Traits> __lhs, basic_string_view<_CharT, _Traits> __rhs) noexcept
+{
+    return __lhs.compare(__rhs) >= 0;
+}
+
+template <class _CharT, class _Traits>
+constexpr bool operator>=(
+    basic_string_view<_CharT, _Traits> __lhs,
+    typename std::common_type<basic_string_view<_CharT, _Traits>>::type __rhs) noexcept
+{
+    return __lhs.compare(__rhs) >= 0;
+}
+
+template <class _CharT, class _Traits>
+constexpr bool operator>=(
+    typename std::common_type<basic_string_view<_CharT, _Traits>>::type __lhs,
+    basic_string_view<_CharT, _Traits> __rhs) noexcept
+{
+    return __lhs.compare(__rhs) >= 0;
+}
+
+typedef basic_string_view<char> string_view;
+typedef basic_string_view<char16_t> u16string_view;
+typedef basic_string_view<char32_t> u32string_view;
+typedef basic_string_view<wchar_t> wstring_view;
+
+inline namespace literals
+{
+    inline namespace string_view_literals
+    {
+        inline basic_string_view<char> operator"" _sv(const char *__str, size_t __len) noexcept
+        {
+            return basic_string_view<char>(__str, __len);
+        }
+
+        inline basic_string_view<wchar_t> operator"" _sv(const wchar_t *__str, size_t __len) noexcept
+        {
+            return basic_string_view<wchar_t>(__str, __len);
+        }
+
+        inline basic_string_view<char16_t> operator"" _sv(const char16_t *__str, size_t __len) noexcept
+        {
+            return basic_string_view<char16_t>(__str, __len);
+        }
+
+        inline basic_string_view<char32_t> operator"" _sv(const char32_t *__str, size_t __len) noexcept
+        {
+            return basic_string_view<char32_t>(__str, __len);
+        }
+    } // namespace string_view_literals
+
+} // namespace literals
+
+// [string.view.hash]
+namespace std
+{
+    template <class _CharT, class _Traits>
+    struct hash<basic_string_view<_CharT, _Traits>> : public unary_function<basic_string_view<_CharT, _Traits>, size_t>
+    {
+        size_t operator()(const basic_string_view<_CharT, _Traits> __val) const noexcept;
+    };
+
+    template <class _CharT, class _Traits>
+    size_t hash<basic_string_view<_CharT, _Traits>>::operator()(const basic_string_view<_CharT, _Traits> __val) const
+        noexcept
+    {
+        auto str = std::basic_string<_CharT, _Traits>(__val.data(), __val.size());
+        return std::hash<std::basic_string<_CharT, _Traits>>()(str);
```

Comment:
This is not ideal, even strange when using string in string_view. But to keep the hash of a basic_string and basic_string_view the same when underlying char sequence are the same, this is the best choice I can think of currently.